lib/SILOptimizer/Mandatory/DataflowDiagnostics.cpp

```diff
@@ -117,35 +119,62 @@ static void diagnoseStaticReports(const SILInstruction *I,
 // SWIFT_ENABLE_TENSORFLOW
 /// \brief Emit a diagnostic for `poundAssert` builtins whose condition is
 /// false or whose condition cannot be evaluated.
-/// TODO(clattner / marcrasi): Implement this for real.
-static void diagnosePoundAssert(const SILInstruction *I, SILModule &M) {
+static void diagnosePoundAssert(const SILInstruction *I, SILModule &M,
+                                ConstExprEvaluator &constantEvaluator) {
   if (auto *builtinInst = dyn_cast<BuiltinInst>(I)) {
```

Comment:
Please make this early-return to reduce indentation, e.g.:

auto *builtinInst = dyn_cast<BuiltinInst>(I)
if (!builtinInst || builtinInst->getKind != ...) return;